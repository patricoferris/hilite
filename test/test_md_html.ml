let md =
  {|
```ocaml
let html = "<h1>Hello World!</h1>"
```

 - ```ocaml
   let indented_code_block f = f
   ```

```diff
-  module Sel = (val if is_osx () then (module Mac)
-                    else if Sys.unix then (module Unix)
-                    else (module Fail) : Unix_socket)
+  module Sel =
+    (val if is_osx () then (module Mac)
+         else if Sys.unix then (module Unix)
+         else (module Fail)
+        : Unix_socket)
```
|}

let test_transform () =
  let omd_t = Cmarkit.Doc.of_string md |> Hilite.Md.transform in
  Format.printf "%s" (Cmarkit_html.of_doc ~safe:false omd_t)

let () = test_transform ()
