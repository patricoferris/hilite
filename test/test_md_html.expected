<pre><code><span class='ocaml-keyword'>let</span><span class='ocaml-source'> </span><span class='ocaml-entity-name-function-binding'>html</span><span class='ocaml-source'> </span><span class='ocaml-keyword-operator'>=</span><span class='ocaml-source'> </span><span class='ocaml-string-quoted-double'>&quot;</span><span class='ocaml-string-quoted-double'>&lt;h1&gt;Hello World!&lt;/h1&gt;</span><span class='ocaml-string-quoted-double'>&quot;</span><span class='ocaml-source'>
</span></code></pre>
<ul>
<li>
<pre><code><span class='ocaml-keyword'>let</span><span class='ocaml-source'> </span><span class='ocaml-entity-name-function-binding'>indented_code_block</span><span class='ocaml-source'> </span><span class='ocaml-source'>f</span><span class='ocaml-source'> </span><span class='ocaml-keyword-operator'>=</span><span class='ocaml-source'> </span><span class='ocaml-source'>f</span><span class='ocaml-source'>
</span></code></pre>
</li>
</ul>
<pre><code><span class='diff-punctuation-definition-deleted'>-</span><span class='diff-markup-deleted'>  module Sel = (val if is_osx () then (module Mac)
</span><span class='diff-punctuation-definition-deleted'>-</span><span class='diff-markup-deleted'>                    else if Sys.unix then (module Unix)
</span><span class='diff-punctuation-definition-deleted'>-</span><span class='diff-markup-deleted'>                    else (module Fail) : Unix_socket)
</span><span class='diff-punctuation-definition-inserted'>+</span><span class='diff-markup-inserted'>  module Sel =
</span><span class='diff-punctuation-definition-inserted'>+</span><span class='diff-markup-inserted'>    (val if is_osx () then (module Mac)
</span><span class='diff-punctuation-definition-inserted'>+</span><span class='diff-markup-inserted'>         else if Sys.unix then (module Unix)
</span><span class='diff-punctuation-definition-inserted'>+</span><span class='diff-markup-inserted'>         else (module Fail)
</span><span class='diff-punctuation-definition-inserted'>+</span><span class='diff-markup-inserted'>        : Unix_socket)
</span></code></pre>
